src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -277,15 +277,28 @@ public void LogAnalyzerCountSummary()
             ResetDiagnosticLogAggregator();
         }
 
-        private void ResetDiagnosticLogAggregator()
+        public static async Task<IEnumerable<DiagnosticData>> GetDiagnosticsAsync(
+            IDiagnosticAnalyzerService service, Document document, IEnumerable<DiagnosticAnalyzer> analyzers, AnalysisKind kind, CancellationToken cancellationToken)
         {
-            DiagnosticLogAggregator = new DiagnosticLogAggregator(Owner);
+            // given service must be DiagnosticAnalyzerService
+            var diagnosticService = (DiagnosticAnalyzerService)service;
```

Comment:
then why not type things to enforce that?  i.e. don't take in an IDiagnosticAnalyzerService